demos/common/include/samples/images_capture.h

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <memory>
+#include <string>
+
+#include <opencv2/core/mat.hpp>
+#include <opencv2/videoio.hpp>
+
+class ImagesCapture {
+public:
+    const bool loop;
+
+    ImagesCapture(bool loop) : loop{loop} {}
+    virtual double fps() const = 0;
+    virtual size_t lastImgId() const = 0;
+    virtual cv::Mat read() = 0;
+    virtual ~ImagesCapture() = default;
+};
+
+// This is a more advanced version of
+// try {
+//     return cv::VideoCapture(std::stoi(input));
+// } catch (const std::invalid_argument&) {
+//     return cv::VideoCapture(input);
+// } catch (const std::out_of_range&) {
+//     return cv::VideoCapture(input);
+// }
+std::unique_ptr<ImagesCapture> openImagesCapture(const std::string &input,
+        bool loop, size_t posFrames=0,  // Non camera options
+        size_t readLengthLimit=std::numeric_limits<size_t>::max(),  // general option
+        // Camera options:
+        double buffersize=1, double frameWidth=1280, double frameHeight=720, double autofocus=true,
+        double fourcc=cv::VideoWriter::fourcc('M', 'J', 'P', 'G'));
```

Comment:
It's useful to allow changing `frameWidth` and `frameHeight` to save some computation on resizes if needed. text_detection may disable `autofocus`. There is a python demo with different `fourcc` (but I'm not sure it really should be different). `buffersize` is the only option which is usually set by demos unconditionally (if set). I made `buffersize` part of the interface just to export all the options that are going to be changed.

> I would suggest to combine the camera options into a struct. That would make the calls more readable.

Function calls will end up with just an extra pair of brackets like `openImagesCapture(FLAGS_i, FLAGS_loop, FLAGS_first, FLAGS_limit, {1, 1280, 720, true, cv::VideoWriter::fourcc('M', 'J', 'P', 'G')})`

Reply:
> It's useful to allow changing frameWidth and frameHeight to save some computation on resizes if needed.

That's fair enough, but that only helps if you actually set those values to the input size of the network. You're not doing that in this PR; you always set the frame size to 1280x720, which seems arbitrary and unnecessary.

> text_detection may disable autofocus.

Okay.

> There is a python demo with different fourcc (but I'm not sure it really should be different).

Python demos aren't going to be using this class.

> buffersize is the only option which is usually set by demos unconditionally (if set). I made buffersize part of the interface just to export all the options that are going to be changed.

That doesn't make sense. A parameter is only needed if it can have different values.

> Function calls will end up with just an extra pair of brackets

Only if you write them that way. What I had in mind was something like this:

```c++
CameraOptions cameraOptions;
cameraOptions.frameSize = cv::Size(...);
capture = openImagesCapture(FLAGS_i, FLAGS_loop, FLAGS_first, FLAGS_limit, cameraOptions);
```

(The parameters that are not set explicitly would be initialized by default.)

Reply:
Updated